src/main/java/org/folio/oaipmh/mappers/MarcXmlMapper.java

```diff
@@ -24,33 +22,26 @@
    * Convert MarcJson to MarcXML.
    *
    * @param source String representation of MarcJson source.
-   * @return DOM's Node representation of MarcXML
+   * @return byte[] representation of MarcXML
    */
-  public Node convert(String source) {
+  public byte[] convert(String source) {
     try (InputStream inputStream
            = new ByteArrayInputStream(source.getBytes(StandardCharsets.UTF_8))) {
       MarcReader marcJsonReader = new MarcJsonReader(inputStream);
-      DOMResult domResult = new DOMResult();
-      MarcWriter marcXmlWriter = new MarcXmlWriter(domResult);
-      while (marcJsonReader.hasNext()) {
-        Record record = marcJsonReader.next();
+      ByteArrayOutputStream out = new ByteArrayOutputStream();
+      Record record = marcJsonReader.next();
 
-        /*
-         * Fix indicators which comes like "ind1": "\\" in the source string and values are converted to '\'
-         * which contradicts to the MARC21slim.xsd schema. So replacing unexpected char by space
-         */
-        for (DataField data : record.getDataFields()) {
-          if (data.getIndicator1() == '\\') {
-            data.setIndicator1(' ');
-          }
-          if (data.getIndicator2() == '\\') {
-            data.setIndicator2(' ');
-          }
+      for (DataField data : record.getDataFields()) {
```

Comment:
The comments here were actually somewhat helpful... you should consider keeping them.  

Also, fixing up the two indicator fields isn't good enough here.  The backslash is used in place of a space through the MARC record.  Converting these to slashes only in some places of the record introduces inconsistencies.  Doing a global search/replace on the entire record would produce a much cleaner result.

Reply:
Done